cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -297,6 +297,174 @@ func (s *Server) GetAvailablePackageVersions(ctx context.Context, request *corev
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("unable to get kapp-controller package metadatas: %v", err))
+	}
+
+	pkgGVR := schema.GroupVersionResource{Group: packageGroup, Version: packageVersion, Resource: packageResources}
+	pkgs, err := client.Resource(pkgGVR).Namespace(namespace).List(ctx, metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.refName=%s", identifier),
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("unable to list kapp-controller packages: %v", err))
+	}
+	pkgVersions, err := pkgVersionsMap(pkgs.Items)
+	if err != nil {
+		return nil, err
+	}
+	requestedPkgVersion := request.GetPkgVersion()
+	if requestedPkgVersion != "" {
+		// Ensure the version is available.
+		found := false
+		for _, v := range pkgVersions[identifier] {
+			if v.version.String() == requestedPkgVersion {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return nil, status.Errorf(codes.NotFound, fmt.Sprintf("unable to find %q package with version %q", identifier, requestedPkgVersion))
+		}
+	} else {
+		// If the pkgVersion wasn't specified, grab the packages to find the latest.
+		if len(pkgVersions[identifier]) > 0 {
+			requestedPkgVersion = pkgVersions[identifier][0].version.String()
+		} else {
+			return nil, status.Errorf(codes.NotFound, fmt.Sprintf("unable to find any versions for the package %q", identifier))
+		}
+	}
+
+	detail, err := AvailablePackageDetailFromUnstructured(pkgMetadata, requestedPkgVersion, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev1.GetAvailablePackageDetailResponse{
+		AvailablePackageDetail: detail,
+	}, nil
+}
+
+func AvailablePackageDetailFromUnstructured(pkgMeta *unstructured.Unstructured, version, cluster string) (*corev1.AvailablePackageDetail, error) {
+	detail := &corev1.AvailablePackageDetail{}
+
+	// https://carvel.dev/kapp-controller/docs/latest/packaging/#package-metadata
+	name, found, err := unstructured.NestedString(pkgMeta.Object, "metadata", "name")
+	if err != nil || !found || name == "" {
+		return nil, status.Errorf(codes.Internal, "required field metadata.name not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	namespace, found, err := unstructured.NestedString(pkgMeta.Object, "metadata", "namespace")
+	if err != nil || !found {
+		return nil, status.Errorf(codes.Internal, "required field metadata.namespace not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	detail.Name = name
+	displayName, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "displayName")
+	if err != nil || !found {
+		return nil, status.Errorf(codes.Internal, "required field spec.displayName not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	detail.DisplayName = displayName
+
+	// Carvel uses base64-encoded SVG data for IconSVGBase64, whereas we need
```

Comment:
I guess it is always like that? Or someone as a package maintainer can opt to just put there a URL ?   (I need to refresh my carvel knowledge...)

Reply:
The actual field name on the API is `IconSVGBase64`, so looks like it is always required in that form (which makes sense if you want to support on-prem use with limited egress).